cncd/pipeline/pipeline/frontend/yaml/constraint.go

```diff
@@ -163,3 +171,72 @@ func (c *ConstraintMap) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	}
 	return nil
 }
+
+// UnmarshalYAML unmarshals the constraint.
+func (c *ConstraintPath) UnmarshalYAML(value *yaml.Node) error {
+	var out1 = struct {
+		Include libcompose.Stringorslice
+		Exclude libcompose.Stringorslice
+	}{}
+
+	var out2 libcompose.Stringorslice
+
+	err1 := value.Decode(&out1)
+	err2 := value.Decode(&out2)
+
+	c.Exclude = out1.Exclude
+	c.Include = append(
+		out1.Include,
+		out2...,
+	)
+
+	if err1 != nil && err2 != nil {
+		y, _ := yaml.Marshal(value)
+		return fmt.Errorf("Could not parse condition: %s", y)
+	}
+
+	return nil
+}
+
+func (c *ConstraintPath) Match(v []string, message string) bool {
+	// ignore file pattern matches if the commit message contains all
+	if strings.Contains(message, "[ALL]") {
```

Comment:
I'm not sure if we should go with this hard-coded value ...

Reply:
I know what you say. should I go the `[SKIP CI]` way, make it a case insensitive regex? or you don't want it at all in this iteration?
https://github.com/woodpecker-ci/woodpecker/blob/bfc4aa805926d73730cc4c7c1f529993099ef2dd/server/hook.go#L43

Reply:
@6543 got your comment, here is the change: https://github.com/woodpecker-ci/woodpecker/pull/174/commits/7c95fb2ad7b8be67ba7947790bd51674c86f25c4